### PR TITLE
Add mailgun support

### PR DESCRIPTION
**Pull request by markwbrown**
Adds support to send update checks out via mailgun instead of using mailx/ssmtp. Adds 3 args: g, a, & d. -g accepts an e-mail address and triggers send_email_via_mailgun(). -a accepts the api key for mailgun. -d accepts the domain (i.e. mg.yourdomain.com)

### CODE REVIEW

File: update-ngxblocker [hunk 1]
```diff
@@ -272,16 +280,23 @@ send_email() {
...
+	sed -i 's/\x1b\[[0-9;]*m//g' $EMAIL_REPORT
+	report="$(cat $EMAIL_REPORT)"
+	curl -s --user api:$MG_API_KEY https://api.mailgun.net/v3/$MG_DOMAIN/messages -F from='botblocker <botblocker@viti.io>' -F to=$EMAIL -F subject='Nginx Bad Bot Blocker Updated' -F text="$report"
+}
 
 get_options() {
 	local arg= opts=
 
-	while getopts :c:b:i:r:e:m:lnovqh opts "$@"
+	while getopts :c:b:i:r:e:g:a:d:m:lnovqh opts "$@"
```
Comment (itoffshore) [inline update-ngxblocker]: Add option for Mailgun FROM address

`while getopts :c:b:i:r:e:g:a:d:f:m:lnovqh opts "$@"`

File: update-ngxblocker [hunk 2]
```diff
@@ -308,7 +326,7 @@ get_options() {
 }
 
 main() {
-	local REPO=https://raw.githubusercontent.com/mitchellkrogza/nginx-ultimate-bad-bot-blocker/master
+	local REPO=https://raw.githubusercontent.com/markwbrown/nginx-ultimate-bad-bot-blocker/master
```
Comment (itoffshore) [inline update-ngxblocker]: Please `rebase` both commits in this `PR` into a single commit

you can see a [rebasing example on my site](https://it-offshore.co.uk/code/81-alpine-linux-pull-requests-on-github)

File: update-ngxblocker [hunk 3]
```diff
@@ -326,7 +326,7 @@ get_options() {
 }
 
 main() {
-	local REPO=https://raw.githubusercontent.com/markwbrown/nginx-ultimate-bad-bot-blocker/master
+	local REPO=https://raw.githubusercontent.com/mitchellkrogza/nginx-ultimate-bad-bot-blocker/master
```
Comment (itoffshore) [inline update-ngxblocker]: Please `rebase` both commits in this `PR` into a single commit

you can see a [rebasing example on my site](https://it-offshore.co.uk/code/81-alpine-linux-pull-requests-on-github)

File: update-ngxblocker [hunk 4]
```diff
@@ -56,6 +60,9 @@ Usage: $script [OPTIONS]
         [ -i ] : Change installer path         (default: $INSTALLER)
         [ -r ] : Change repo url               (default: $REPO)
         [ -e ] : Change @email address         (default: $EMAIL)
+        [ -g ] : Change @email address mailgun         (default: $EMAIL)
```
[the remark this answers is not in the document]
Comment (markwbrown) [reply update-ngxblocker]: Capitalized

File: update-ngxblocker [hunk 5]
```diff
@@ -56,6 +60,9 @@ Usage: $script [OPTIONS]
...
+        [ -d ] : Set Domain
```
[the remark this answers is not in the document]
Comment (markwbrown) [reply update-ngxblocker]: Capitalized

File: update-ngxblocker [hunk 6]
```diff
@@ -85,13 +93,13 @@ check_version() {
 		date=$(grep "Updated:" $file | sed 's|^.*: ||g')
 		print_message "\nLOCAL Version: $BOLDWHITE$version$RESET\n"
 		print_message "Updated: $date\n\n"
-
```
[the remark this answers is not in the document]
Comment (markwbrown) [reply update-ngxblocker]: Accidental

File: update-ngxblocker [hunk 7]
```diff
@@ -85,13 +93,13 @@ check_version() {
...
 		# remote version
 		curl -s --limit-rate 5k -r $range --location $url -o $tmp
 		remote_ver=$(grep "Version:" $tmp | sed 's|^.*: V||g')
 		remote_date=$(grep "Updated:" $tmp | sed 's|^.*: ||g')
 		print_message "REMOTE Version: $BOLDWHITE$remote_ver$RESET\n"
 		print_message "Updated: $remote_date\n"
+
```
[the remark this answers is not in the document]
Comment (markwbrown) [reply update-ngxblocker]: Accidental.

File: update-ngxblocker [hunk 8]
```diff
@@ -30,7 +30,11 @@
 
 EMAIL="you@example.com"
 SEND_EMAIL="N"
+SEND_MG_EMAIL="N"
 SEND_EMAIL_UPDATE="N"
+#Mailgun
+MG_API_KEY="key-yadayadayada"
+MG_DOMAIN="mg.example.com"
```
[the remark this answers is not in the document]
Comment (markwbrown) [reply update-ngxblocker]: Added $MG_FROM

File: update-ngxblocker [hunk 9]
```diff
@@ -272,16 +280,23 @@ send_email() {
 		print_message "${BOLDYELLOW}WARN${RESET}: missing mail command => ${BOLDWHITE}disabling emails${RESET}.\n\n"
 	fi
 }
+send_email_via_mailgun() {
+	echo "Mailgunning report to: ${BOLDWHITE}$EMAIL${RESET}\n\n";
+	sed -i 's/\x1b\[[0-9;]*m//g' $EMAIL_REPORT
+	report="$(cat $EMAIL_REPORT)"
+	curl -s --user api:$MG_API_KEY https://api.mailgun.net/v3/$MG_DOMAIN/messages -F from='botblocker <botblocker@viti.io>' -F to=$EMAIL -F subject='Nginx Bad Bot Blocker Updated' -F text="$report"
```
[the remark this answers is not in the document]
Comment (markwbrown) [reply update-ngxblocker]: added $MG_FROM

File: update-ngxblocker [hunk 10]
```diff
@@ -292,6 +307,9 @@ get_options() {
 			i) INSTALLER=$arg; check_args $opts script $arg ;;
 			r) REPO=$arg; check_args $opts url $arg ;;
 			e) EMAIL=$arg; SEND_EMAIL=Y; check_args $opts email $arg ;;
+			g) EMAIL=$arg; SEND_MG_EMAIL=Y; check_args $opts email $arg ;;
+			a) MG_API_KEY=$arg;; 
+			d) MG_DOMAIN=$arg;;
```
[the remark this answers is not in the document]
Comment (markwbrown) [reply update-ngxblocker]: added MG_FROM

File: update-ngxblocker (hunk 1 above)
Comment (markwbrown) [reply update-ngxblocker]: yup.

File: update-ngxblocker (hunk 2 above)
Comment (markwbrown) [reply update-ngxblocker]: rebased changes

**Comment by markwbrown** [comment]
@itoffshore Changes are up. It's a bit ugly with the conflicts-- it looks like updates were introduced by @mitchellkrogza

**Comment by itoffshore** [comment]
@markwbrown - you have made the changes on your `master` branch. The `PR` needs to be made from a separate `branch` in your forked `repo` so it can be merged cleanly as mentioned in [my github notes](https://it-offshore.co.uk/code/81-alpine-linux-pull-requests-on-github). I would do something like:

```
cp /path/to/update-ngxblocker-with-changes /some/where/safe

# hard reset of your forked repo
git checkout master
git remote update
git reset --hard upstream/master --
git push origin +master

# work from a branch
git checkout -b mailgun
cp /some/where/safe/update-ngxblocker-with-changes ./update-ngxblocker
```

& make a new clean `PR`. This avoids merge conflicts from changes to `master`.

When I am working on a `branch` & before I `git commit` changes I always check the changes are what I expect with `git diff`

**Comment by itoffshore** [comment]
I also think you should create a function of `send_mailgun_email()` in a similar way to `send_email()`

**Comment by markwbrown** [comment]
> I also think you should create a function of `send_mailgun_email()` in a similar way to `send_email()`

can you clarify? Are you saying rename send_email_via_mailgun() to send_mailgun_email()?

**Comment by itoffshore** [comment]
@markwbrown - if you already have the `mailgun` functionality in a  `send_email_via_mailgun()` function it's ok.

**Comment by markwbrown** [comment]
> @markwbrown - if you already have the `mailgun` functionality in a `send_email_via_mailgun()` function it's ok.

@itoffshore Yes -- that function sends a curl request to mailgun's api, passing in the api key from the -a flag and the domain from the -d flag. That is all that is needed to send.

**Comment by itoffshore** [comment]
@markwbrown - please see my comments above to [fix merging the changes](https://github.com/mitchellkrogza/nginx-ultimate-bad-bot-blocker/pull/261/#issuecomment-500116397).

**Pull request closed without merging**